nntrainer/src/bn_layer.cpp

```diff
@@ -151,30 +159,20 @@ Tensor BatchNormalizationLayer::backwarding(Tensor dy, int iteration) {
          .divide_i(cvar.multiply(batch))
          .run();
 
-  gradients.clear();
-  gradients.push_back(dgamma);
-  gradients.push_back(dbeta);
+  std::shared_ptr<UpdatableParam> grad_params(params, params.get() + 2);
 
-  opt.apply_gradients(weights, gradients, iteration);
+  opt.apply_gradients(grad_params, param_size - 2, iteration);
 
   return dx;
 }
 
-void BatchNormalizationLayer::read(std::ifstream &file) {
-  mu.read(file);
-  var.read(file);
-  gamma.read(file);
-  beta.read(file);
-}
+void BatchNormalizationLayer::read(std::ifstream &file) { Layer::read(file); }
```

Comment:
i think read and save need not be pure virtual now. In another PR possibly.

Reply:
Yes. it is not pure virtual in this PR.

Reply:
So, this definition can be just removed from here?

Reply:
Right. Thanks :)